pkg/installation/install_test.go

```diff
@@ -257,3 +259,41 @@ func Test_downloadAndExtract_fileOverride(t *testing.T) {
 		t.Fatal("no files found in the extract output directory")
 	}
 }
+
+func Test_applyDefaults(t *testing.T) {
+	tests := []struct {
+		name     string
+		platform index.Platform
+		expected index.Platform
+	}{
+		{
+			name:     "with files given",
+			platform: testutil.NewPlatform().WithFiles([]index.FileOperation{{From: "here", To: "there"}}).V(),
+			expected: testutil.NewPlatform().WithFiles([]index.FileOperation{{From: "here", To: "there"}}).V(),
+		},
+		{
+			name:     "with empty files",
+			platform: testutil.NewPlatform().WithFiles([]index.FileOperation{}).V(),
+			expected: testutil.NewPlatform().WithFiles([]index.FileOperation{}).V(),
+		},
+		{
+			name:     "with unspecified files",
+			platform: testutil.NewPlatform().WithFiles(nil).V(),
+			expected: testutil.NewPlatform().WithFiles([]index.FileOperation{{From: "*", To: "."}}).V(),
+		},
+		{
+			name:     "no defaults for other fields",
+			platform: testutil.NewPlatform().WithBin("").WithOS("").WithSelector(nil).WithSHA256("").WithURI("").V(),
```

Comment:
oh boy this test is gonna be a problem in the future.
because we will keep adding defaults to testutil.NewPlatform(), and we will keep adding new WithXxx.
let's keep it until it breaks.